packages/lib/src/components/ThreeDS2/components/Challenge/PrepareChallenge3DS2.tsx

```diff
@@ -48,18 +58,49 @@ class PrepareChallenge3DS2 extends Component<PrepareChallenge3DS2Props, PrepareC
             return (
                 <DoChallenge3DS2
                     onCompleteChallenge={(challenge: ThreeDS2FlowObject) => {
+                        // Challenge has resulted in an error (no transStatus could be retrieved) - but we still treat this as a valid scenario
+                        if (hasOwnProperty(challenge.result, 'errorCode')) {
+                            // Tell the merchant there's been an error
+                            const errorObject = handleErrorCode(challenge.result.errorCode);
+                            this.props.onError(errorObject);
+                            // Proceed with call to onAdditionalDetails
+                            this.setStatusComplete(challenge);
+                            return;
+                        }
+
+                        // A valid result (w. transStatus)
                         this.setStatusComplete(challenge.result);
                     }}
                     onErrorChallenge={(challenge: ThreeDS2FlowObject) => {
-                        const errorObject = handleErrorCode(challenge.errorCode);
-                        this.props.onError(errorObject);
-                        this.setStatusComplete(challenge.result);
+                        // Called when challenge times-out (which is still a valid scenario)
+                        if (hasOwnProperty(challenge, 'errorCode')) {
+                            const errorObject = handleErrorCode(challenge.errorCode);
+                            this.props.onError(errorObject);
+                            this.setStatusComplete(challenge.result);
+                            return;
+                        }
+
+                        // or, when the challenge response is un-parseable JSON - in which case something unknown has gone wrong
+                        this.setStatusError(challenge);
                     }}
                     {...challengeData}
                 />
             );
         }
 
+        if (this.state.status === 'error') {
+            return (
+                <div className="adyen-checkout__threeds2-challenge-error">
+                    <Img
+                        className="adyen-checkout__status__icon adyen-checkout__status__icon--error"
+                        src={getImageUrl({ loadingContext: this.props.loadingContext, imageFolder: 'components/' })('error')}
+                        alt={this.props.i18n.get('error.message.unknown')}
```

Comment:
We don't need alt here since we already use `error.message.unkown` in the text message.

Reply:
Good point!